rotkehlchen/chain/scroll/modules/weth/decoder.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from typing import TYPE_CHECKING
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.decoding.structures import DEFAULT_DECODING_OUTPUT
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.chain.evm.decoding.weth.decoder import WethDecoder as EthBaseWethDecoder
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.structures import DecoderContext, DecodingOutput
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class WethDecoder(EthBaseWethDecoder):
+    def _decode_deposit_event(self, context: 'DecoderContext') -> 'DecodingOutput':
+        depositor = hex_or_bytes_to_address(context.tx_log.topics[1])
+        deposited_amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        deposited_amount = asset_normalized_value(
+            amount=deposited_amount_raw,
+            asset=self.base_asset,
+        )
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.address in (self.wrapped_token.evm_address, depositor) and
+                event.balance.amount == deposited_amount and
+                event.asset == self.base_asset
+            ):
+                if event.address == depositor:
+                    return DEFAULT_DECODING_OUTPUT
+
+                event.counterparty = self.counterparty
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Wrap {deposited_amount} {self.base_asset.symbol} in {self.wrapped_token.symbol}'  # noqa: E501
```

Comment:
Just realized this is the scroll decoder. So you can simplify
```suggestion
                event.notes = f'Wrap {deposited_amount} ETH in WETH'  # noqa: E501
```